tests/testthat/test-theme_elements_gtsummary.R

```diff
@@ -0,0 +1,212 @@
+# Package-wide Unit Tests-------------------------------------------------------
+## pkgwide-fn:prependpvalue_fun-------------------------------------------------
+test_that("pkgwide-fn:prependpvalue_fun works", {
+
+  # Create a theme#
+  my_theme_1 <-
+    list(
+      # Prepend p value, with 3 place digits
+      "pkgwide-fn:prependpvalue_fun" = label_style_pvalue(digits = 3, prepend_p = TRUE)
+    )
+
+  # Set the theme
+  set_gtsummary_theme(my_theme_1)
+
+  # Store the table#
+  gts_1 <-
+    trial |>
+    tbl_summary(
+      by = trt,
+      include = c(trt, age),
+    ) |>
+    add_p()
+
+  # Test that the p-value has 3 digits#
+  expect_true(
+    inline_text(gts_1, variable = age, column = "p.value") |>
+      grepl(pattern = "p=0.\\d{3}", x = _)
+  )
+  # Reset the theming#
+  reset_gtsummary_theme()
+}
+)
+
+## Other pkgwide Tests----------------------------------------------------------
+# Create a theme#
+my_theme_2 <-
+  list(
+    # Prepend p value, with 3 place digits
+    "pkgwide-fn:pvalue_fun" = label_style_pvalue(digits = 1, prefix = "Totally Awesome P Value = ", "OutDec" = ","),
+    # set messaging to quiet
+    "pkgwide-lgl:quiet" = TRUE,
+    # configurar el idioma en español#
+    "pkgwide-str:language" = "es"
+  )
+
+# Set the theme
+set_gtsummary_theme(my_theme_2)
```

Comment:
Use `with_gtsummary_theme()` as above

Reply:
Ok so when I try to use `with_gtsummary_theme()` to test `"pkgwide-fn:pvalue_fun"` I'm running into issues:

``` r
# Library Load-in---------------------------------------------------------------
library(gtsummary)
library(testthat)
#> Warning: package 'testthat' was built under R version 4.3.3
#> 
#> Attaching package: 'testthat'
#> The following object is masked from 'package:gtsummary':
#> 
#>     matches

## Other pkgwide Tests----------------------------------------------------------
# Create a theme#
my_theme_2 <-
  list(
    # Pvalue with 2 place digits and custom decimal mark
    "pkgwide-fn:pvalue_fun" = label_style_pvalue(digits = 2, decimal.mark = "++"),
    # set messaging to quiet
    "pkgwide-lgl:quiet" = TRUE,
    # configurar el idioma en español#
    "pkgwide-str:language" = "es"
  )

# Store the table#
gts_2 <-
  trial |>
  tbl_summary(
    by = trt,
    include = c(trt, age),
  ) |>
  add_p()

# Apply the theme to the table#
gts_2_pvalue <-
  with_gtsummary_theme(
    x = my_theme_2,
    expr =  inline_text(gts_2, column = "p.value", variable = age) # Does not work as expected
  )

# Test that the p-value has the decimal mark - fails#
expect_equal(gts_2_pvalue, "0++72")
#> Error: `gts_2_pvalue` not equal to "0++72".
#> 1/1 mismatches
#> x[1]: "p=0.7"
#> y[1]: "0++72"
```


But, `"pkgwide-fn:prependpvalue_fun"` works as I would expect:
```
# Package-wide Unit Tests-------------------------------------------------------
## pkgwide-fn:prependpvalue_fun-------------------------------------------------
  
  # Create a theme#
  my_theme_1 <-
    list(
      # Prepend p value, with 3 place digits
      "pkgwide-fn:prependpvalue_fun" = label_style_pvalue(digits = 3, prepend_p = TRUE)
    )
  
  # Create a summary table#
  gts_1 <-
    trial |>
    tbl_summary(
      by = trt,
      include = c(trt, age)
    ) |>
    add_p()
  
  # Apply the theme to the table#
  gts_1_pvalue <-
    with_gtsummary_theme(
      x = my_theme_1,
      expr = inline_text(x = gts_1, variable = age, column = "p.value")   # Works as expected
    )
  
  # Test that the p-value has 3 digits - passes#
  expect_equal(gts_1_pvalue, "p=0.718")

```

Reply:
The `"pkgwide-fn:pvalue_fun"` doesn't control the default function in `inline_text()`, but it does control the default in `add_p()`. I think that is where the difference is coming from.

Reply:
I got it work by using `inline_text.gtsummary()` and by not storing the table in an object before passing it through to `with_gtsummary_theme()`. I'll let you know if I run into anything else. Thank you!